autogen/variables.tf

```diff
@@ -300,6 +300,34 @@ variable "pod_security_policy_config" {
     "enabled" = false
   }]
 }
+
+variable "authenticator_groups_config" {
+  description = "Configuration for the Google Groups for GKE feature.  This block has a field named 'security_group' which lists RBAC security group for use with Google security groups in Kubernetes RBAC."
+  default = []
+}
+
+
+variable "vertical_pod_autoscaling" {
+  description = "Vertical Pod Autoscaling automatically adjusts the resources of pods controlled by it."
+  default     = [{
+    "enabled" = false
+  }]
+}
+
+variable "workload_identity_config" {
```

Comment:
I can't even find an explanation of the expected value here in the docs.

We should either (a) expose the underlying options as separate variables or (b) provide an example of what the input list should look like.

Reply:
The docs are a bit lacking at times.  From values in the source code I was able to map it back to a flag in gcloud:

```
     --workload-metadata-from-node=WORKLOAD_METADATA_FROM_NODE
        Type of metadata server available to pods running in the nodepool.
        WORKLOAD_METADATA_FROM_NODE must be one of:

         EXPOSED
            Pods running in this nodepool have access to the node's underlying
            Compute Engine Metadata Server.
         GKE_METADATA_SERVER
            Run the Kubernetes Engine Metadata Server on this node. The
            Kubernetes Engine Metadata Server exposes a metadata API to
            workloads that is compatible with the V1 Compute Metadata APIs
            exposed by the Compute Engine and App Engine Metadata Servers. This
            feature can only be enabled if Workload Identity is enabled at the
            cluster level.
         SECURE
            Prevents pods not in hostNetwork from accessing certain VM
            metadata, specifically kube-env, which contains Kubelet
            credentials, and the instance identity token. This is a temporary
            security solution available while the bootstrapping process for
            cluster nodes is being redesigned with significant security
            improvements. This feature is scheduled to be deprecated in the
            future and later removed.
         UNSPECIFIED
            Chooses the default.
```

Reply:
Yeah, let's try to include those in the docs.

Reply:
I'm not sure that is correct @chrislovecnm ?

Those values are intended for setting `nodeMetadata` in [node pool config](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/NodeConfig#workloadmetadataconfig) whereas for the `identityNamespace` for [WorkloadIdentityConfig](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#Cluster.WorkloadIdentityConfig) it should be in the format of 

https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_a_new_cluster
```bash
gcloud beta container clusters create [CLUSTER_NAME] \
  --cluster-version=1.12 \
  --identity-namespace=[PROJECT_NAME].svc.id.goog
```

So for terraform its:
```terraform
  workload_identity_config {
    identity_namespace = "${var.project_id}.svc.id.goog"
  }
```